src/controllers/auth.controller.js

```diff
@@ -0,0 +1,262 @@
+const bcrypt = require('bcrypt');
+const uuid = require('uuid');
+
+require('dotenv').config();
+
+const emailService = require('../services/email.service');
+const userService = require('../services/user.service');
+const jwtService = require('../services/jwt.service');
+const tokenService = require('../services/token.service');
+const { ApiError } = require('../utils/api.error');
+const { User } = require('../models/user.model');
+
+const register = async (req, res) => {
+  const { name, email, password } = req.body;
+  const activationToken = uuid.v4();
```

Comment:
Consider using a separate token for activation and password reset to avoid potential security risks. Using the same token type for different purposes can lead to vulnerabilities.